ethcore/sync/src/chain/mod.rs

```diff
@@ -672,6 +703,8 @@ pub struct ChainSync {
 	/// Connected peers pending Status message.
 	/// Value is request timestamp.
 	handshaking_peers: HashMap<PeerId, Instant>,
+	/// Requests, that can not be processed at the moment
+	delayed_requests: HashSet<(PeerId, u8, Vec<u8>)>,
```

Comment:
Shouldn't we maintain ordering of the requests? With `HashSet` they will get randomized.

Reply:
Reworked to vec. Standard option with linked_hash_map doesn't suit because of MallocSizeOf bound